File: src/main/java/com/cqjtu/cssl/entity/LabInfo.java
package com.cqjtu.cssl.entity;

import com.baomidou.mybatisplus.annotation.TableId;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.experimental.Accessors;

import java.io.Serializable;
import java.math.BigDecimal;

/**
 * 实验室信息实体类
 *
 * @author suwen Aplin
 * @since 2020-02-27
 */
@ApiModel(description = "实验室信息实体")
@Data
@EqualsAndHashCode(callSuper = false)
@Accessors(chain = true)
public class LabInfo implements Serializable {

  private static final long serialVersionUID = 1L;

  @ApiModelProperty(value = "实验室编号", required = true)
  @TableId
  private String labId;

  @ApiModelProperty(value = "实验室负责人ID", position = 1, required = true)
  private String tid;

  @ApiModelProperty(value = "实验室类型ID", position = 2, required = true)
  private Integer typeId;

  @ApiModelProperty(value = "实验室名称", position = 3, required = true)
  private String labName;

  @ApiModelProperty(value = "校区", position = 4, required = true)
  private String labCampus;

  @ApiModelProperty(value = "人数容量", position = 5, required = true)
  private Integer labCap;

  @ApiModelProperty(value = "实验室面积", position = 6, required = true)
  private BigDecimal labArea;
}

File: src/app/enity/labInfo.ts
/**
 * 实验室信息实体
 *
 * @author suwen
 * @date 2020/8/21 下午1:36
 */
export class LabInfo {
  /** 实验室编号 */
  labId: string;

  /** 实验室负责人ID */
  tid: string;

  /** 实验室类型ID */
  typeId: number;

  /** 实验室名称 */
  labName: string;

  /** 校区 */
  labCampus: string;

  /** 人数容量 */
  labCap: number;

  /** 实验室面积 */
  labArea: number;

  /** 实验室负责人名 */
  tname: string;

  /** 实验室类型 */
  typeName: string;

  constructor() {
    this.labId = '';
    this.tid = '';
    this.typeId = 0;
    this.labName = '';
    this.labCampus = '';
    this.labCap = 0;
    this.labArea = 0;
    this.tname = '';
    this.typeName = '';
  }
}
